lib/src/solver/version_solver.dart

```diff
@@ -80,29 +79,7 @@ class VersionSolver {
   VersionSolver(this._type, this._systemCache, this._root, this._lockFile,
       Iterable<String> useLatest)
       : _overriddenPackages = new MapKeySet(_root.pubspec.dependencyOverrides),
-        _useLatest = new Set.from(useLatest) {
-    _addImplicitIncompatibilities();
-  }
-
-  /// Adds incompatibilities representing the dependencies pub itself has on
-  /// various packages to support barback at runtime.
-  void _addImplicitIncompatibilities() {
-    var barbackOverride = _root.pubspec.dependencyOverrides["barback"];
-    var barbackRef = barbackOverride == null
-        ? _systemCache.sources.hosted.refFor("barback")
-        : barbackOverride.toRef();
-
-    pubConstraints.forEach((name, constraint) {
-      if (_root.pubspec.dependencyOverrides.containsKey(name)) return;
-
-      _addIncompatibility(new Incompatibility([
-        new Term(barbackRef.withConstraint(VersionConstraint.any), true),
-        new Term(
-            _systemCache.sources.hosted.refFor(name).withConstraint(constraint),
-            false)
-      ], IncompatibilityCause.pubDependency));
```

Comment:
Please also remove `IncompatibilityCause.pubDependency` and its other referents.